tools/exportNotebook

```diff
@@ -4,37 +4,78 @@
 # - convert (imagemagick)
 # - pdftk (pdftk)
+# - rsvg-convert (optional, to avoid rasterizing of lines)
+
+if [[ $# -eq 0 ]] ; then
+    echo "Usage: ./exportNotebook (Partial)NotebookName AdditionalrM2svgArguments"
+    echo "You can additionally append the -c argument for coloured annotations."
+    exit 0
+fi
+
+# Check if ssh configuration for "remarkable" exists
+grep -Fxq "host remarkable" ~/.ssh/config
+if [ $? -eq 0 ]; then
+    SSH_IP="root@remarkable"
+else
+    SSH_IP="root@10.11.99.1"
+fi
 
 # Getting the notebook prefix (Newest notebook matching the name)
-id=$(ssh root@10.11.99.1 "ls -rt .local/share/remarkable/xochitl/*.metadata | xargs fgrep -l $1" | tail -n1 | cut -d. -f1,2)
+id=$(ssh ${SSH_IP} "ls -rt .local/share/remarkable/xochitl/*.metadata | xargs fgrep -l $1" | tail -n1 | cut -d. -f1,2)
 
 test -z "$id" && exit 1
 
 tmpfolder=$(mktemp -d)
 
 # Getting notebook data
-scp -q root@10.11.99.1:"${id}".{lines,pagedata,metadata} "${tmpfolder}"/
+scp -q ${SSH_IP}:"${id}".{lines,pagedata,metadata} "${tmpfolder}"/
+
+# Copy underyling document pdf if it exists
+ssh -q ${SSH_IP} "[[ -f "${id}.pdf" ]]" && scp -q ${SSH_IP}:"${id}.pdf" "${tmpfolder}"
+
+# Fix for single page notebooks with no template (empty pagedata file by default)
+if [ ! -s "${tmpfolder}"/*.pagedata ]
+then
+  echo "Blank" > "${tmpfolder}"/*.pagedata
+fi
 
 # Fix empty lines in pagedata files
 sed -i -e "s/^[[:blank:]]*$/Blank/" "${tmpfolder}"/*.pagedata
 
 filename=$(grep -F  '"visibleName"' "${tmpfolder}"/*.metadata | cut -d: -f2- | grep -o '"[^"]*"')
 echo "Exporting notebook ${filename} ($(wc -l "${tmpfolder}"/*.pagedata | cut -d\  -f1) pages)"
 
-# Getting template files
-sort -u "${tmpfolder}"/*.pagedata | while read -r tpl; do
-  scp -q root@10.11.99.1:"'/usr/share/remarkable/templates/${tpl}.png'" "${tmpfolder}"/
-done
+if [ -f "${tmpfolder}"/*.pdf ]
+then
+    echo "Found underlying document PDF, using as background."
+    cp "${tmpfolder}/"*.pdf "${tmpfolder}/background.pdf"
+else
+    # Getting template files
+    sort -u "${tmpfolder}"/*.pagedata | while read -r tpl; do
+    scp -q ${SSH_IP}:"'/usr/share/remarkable/templates/${tpl}.png'" "${tmpfolder}"/
+    done
 
-# Generate a PDF file out of the templates
-sed -e "s|^|\"${tmpfolder}\"/\"|" -e 's|$|.png"|' "${tmpfolder}"/*.pagedata | tr '\n' ' ' | sed -e "s|$|-transparent white \"${tmpfolder}\"/background.pdf|" | xargs convert
+    # Generate a PDF file out of the templates
+    sed -e "s|^|\"${tmpfolder}\"/\"|" -e 's|$|.png"|' "${tmpfolder}"/*.pagedata | tr '\n' ' ' | sed -e "s|$|-transparent white \"${tmpfolder}\"/background.pdf|" | xargs convert
+fi
 
 # Extract annotations and create a PDF
-rM2svg --input "${tmpfolder}"/*.lines --output "${tmpfolder}/foreground"
-convert -density 100 "${tmpfolder}"/foreground*.svg -transparent white "${tmpfolder}"/foreground.pdf
+rM2svg --input "${tmpfolder}"/*.lines --output "${tmpfolder}/foreground" $2
+
+if type "rsvg-convert" > /dev/null
```

Comment:
Is this the correct condition ? It looks like you need to negate it. `type` returns 0 when the command exists.

Reply:
Well observed, you're absolutely right. Fixed this in https://github.com/reHackable/maxio/pull/7/commits/fb68318bd2c2de09cc15d6fd300552a822ee54d1. Also sorry for mixing commits in different PRs. 

Btw, do you know where the cropping information is stored?

Reply:
As I noted in http://remarkablewiki.com/index.php?title=Filesystem_layout#Content_file_format , the "transform" array in uuid.content is correlated to the zoom level in the crop functionality. However, I couldn't find any persistent storage of the coordinates of the cropped area. This may have changes since the 1.1 update. I have not performed those experiments since.

Reply:
Thanks for the hint, I haven't seen that. I will give it a try.